Rs2Bank and Rs2Prayer Update

slest1 opened the pull request:
## Change Log

### Features

#### Bank Collection Box Interaction (Rs2Bank.java)
- **New methods to manage the Bank Collection Box:**
    - `openCollectionBox()`: Opens the collection box by interacting with the nearest bank, GE booth, or banker.
    - `bankAllCollectionBoxItems()`: Opens the collection box and banks all items.
    - `inventoryAllCollectionBoxItems()`: Opens the collection box and moves all items to inventory.
    - `closeCollectionBox()`: Closes the collection box interface.
    - `collectionBoxIsOpen()`: Checks if the collection box is currently open.

#### Prayer Management (Rs2Prayer.java)
- **New `swapOverHeadPrayer(Rs2PrayerEnum prayer)` method:**
    - Efficiently swaps the active overhead protection prayer. If the target prayer is already active, it does nothing. Otherwise, it deactivates any current overhead prayer and activates the new one.

slest1 committed:
Add collection box interaction methods to Rs2Bank

Implemented methods for opening, closing, and managing the collection box, including actions to deposit or withdraw items. These additions enhance the functionality for handling bank collection boxes in various in-game scenarios.

slest1 committed:
Add method to swap overhead prayers

Introduced a `swapOverHeadPrayer` method to handle switching between protection prayers while disabling the active one, if any.

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java:
```diff
@@ -1215,6 +1210,108 @@ public static boolean openBank() {
+    public static boolean bankAllCollectionBoxItems() {
+        openCollectionBox();
+        sleepUntil(Rs2Bank::collectionBoxIsOpen, 5000);
+        Rs2Widget.clickWidget("Bank");
```
We should fetch the widget object here & click that or define the widget Ids, using clickWidget(string) like this can lead to clicking messages in the chat if there was someone nefarious trying to type "bank" in the chat

slest1 replied:
makes sense. Will update this

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/prayer/Rs2Prayer.java:
```diff
@@ -138,6 +138,17 @@ public static boolean isMeleePrayerActive() {
+        if (activeProtectionPrayer == prayer) {
+            return;
+        }
+        if (activeProtectionPrayer != null) {
```
If we already have a protection prayer enabled & we attempt to switch to a different protection prayer, we don't need to disable the current one as you can only have 1 protection prayer enabled

slest1 replied:
yea oversight on my part, will push an update soon.

slest1 committed:
Refactor widget interactions in Rs2Bank methods

Updated hard-coded widget actions to use specific widget IDs or dynamic children to ensure correct functionality.

slest1 committed:
Cleaned up swapOverHead Prayer

slest1 committed:
Added widget id for inventory Collection box
